test/wasm_test.cc

```diff
@@ -113,4 +114,132 @@ TEST_P(TestVm, GetOrCreateThreadLocalWasmFailCallbacks) {
   ASSERT_NE(thread_local_plugin3->wasm(), thread_local_plugin2->wasm());
 }
 
+// Tests the canary is always applied when making a call `createWasm`
+TEST_P(TestVm, AlwaysApplyCanary) {
+  // Use different root_id, but the others are the same
+  const auto *const plugin_name = "plugin_name";
+
+  const std::string root_ids[2] = {"root_id_1", "root_id_2"};
+  const std::string vm_ids[2] = {"vm_id_1", "vm_id_2"};
+  const std::string vm_configs[2] = {"vm_config_1", "vm_config_2"};
+  const std::string plugin_configs[3] = {"plugin_config_1", "plugin_config_2",
+                                         /* raising the error */ ""};
+  const std::string plugin_keys[2] = {"plugin_key_1", "plugin_key_2"};
+  const auto fail_open = false;
+
+  // Define common callbacks
+  WasmHandleCloneFactory wasm_handle_clone_factory =
```

Comment:
`wasm_handle_clone_factory` isn't used anywhere, and `wasm_handle_clone_factory_for_canary` is used twice. Is this is a leftover from copy&paste or should both versions be used?

Reply:
Ah.. good point. I had to remove this one.